lamindb/curators/__init__.py

```diff
@@ -198,6 +198,12 @@ def save_artifact(
         # constructor signature
         pass
 
+    def _set_dataset_and_artifact(self):
+        if isinstance(self._dataset, Artifact):
+            self._artifact = self._dataset
+            if self._artifact.otype in {"DataFrame", "AnnData"}:
```

Comment:
What do we do in case this evaluates to `False`? We need to raise an error, right?

Reply:
No, we can't here because this will be dealt with in TiledbCurator, SpatialDataCurator etc.